web-app/src/components/Legend/container.js

```diff
@@ -12,152 +12,110 @@ const LegendContainer = Wrapped =>
-      const { takenHolidays } = this.props;
-      const { employees } = this.state;
-      const prevEmployees = prevState.employees;
-
-      if (employees.length === prevEmployees.length && employees.length === 0) {
-        this.createEmployeesList(takenHolidays);
-      }
+    componentWillMount = () => {
+      this.storeLegendKeysToState();
     };
 
-    sortEmployeeList = objArray => {
-      return objArray.sort(function(a, b) {
-        if (a.displayName < b.displayName) return -1;
-        if (a.displayName > b.displayName) return 1;
-        return 0;
-      });
+    componentDidUpdate = () => {
+      const { employeeList } = this.state;
+      if (employeeList.length === 0) {
+        this.storeEmployeeListToState();
+      }
     };
 
-    createEmployeeObject = hol => {
-      let { employeeId, forename, surname } = hol.employee;
-      return {
-        value: employeeId,
-        displayValue: `${forename} ${surname}`,
-      };
-    };
+    storeLegendKeysToState = () => {
+      const statusKeys = Object.keys(holidayStatus);
+      const statusList = [];
+      const typeList = [];
 
-    removeDuplicateEmployees = employees => {
-      return employees.reduce((unique, o) => {
-        if (
-          !unique.some(
-            obj => obj.value === o.value && obj.displayValue === o.displayValue
-          )
-        ) {
-          unique.push(o);
+      for (const index of statusKeys.keys()) {
+        const keyObj = {
+          eventId: index + 1,
+          key: index + 1,
+          active: false,
+        };
+        // temp if statement
+        if (index < 4) {
+          statusList.push(keyObj);
+        } else {
+          typeList.push(keyObj);
         }
-        return unique;
-      }, []);
-    };
+      }
 
-    createEmployeesList = takenHolidays => {
-      let employees = takenHolidays.filter(hol => hol.employee).map(hol => {
-        return this.createEmployeeObject(hol);
+      this.setState({
+        statusList: statusList,
+        typesList: typeList,
       });
-      employees = this.removeDuplicateEmployees(employees);
-      employees.unshift({ value: -1, displayValue: 'All' });
-      this.setState({ employees: this.sortEmployeeList(employees) });
     };
 
-    setKeyActiveState = eventStatusId => {
-      let updatedList = [...this.state.eventsKeys];
+    storeEmployeeListToState = () => {
+      const { takenHolidays } = this.props;
+      const employeeList = takenHolidays
+        .filter(hol => hol.employee)
+        .map(hol => hol.employee)
+        .filter((employee, pos, arr) => {
+          return (
+            arr
+              .map(mapObj => mapObj.employeeId)
+              .indexOf(employee.employeeId) === pos
+          );
+        });
+      this.setState({ employeeList });
+    };
+
+    setLegendKeyActiveState = (eventId, event) => {
+      let updatedList;
+      let updatedProp;
+      if (event === 'status') {
+        updatedList = [...this.state.statusList];
```

Comment:
Rather than maintaining two separate lists for events, I think we should keep it as one and format the data being passed into our Wrapped component. It would make this function a lot cleaner/clearer.

Also, it might be worth looking into whether or not we need this complexity around these keyObjects and lists. Would it not be easier to maintain a single array of active event filter IDs or something? The fact that the event statuses/types are a constant piece of data, I kinda feel like we could be smarter about how we generate/manage the components.

Also also, it's 22:34 and this is just me thinking out loud.